_kbarticlespages/repository-best-practices.md

```diff
@@ -0,0 +1,67 @@
+
+#### **Private**
+
+##### **Pros**
+  - Restrict access to authorized collaborators and contributors (for any support issues a Pages engineer will request to be added as a collaborator to the repository if necessary)
+  - Risk mitigation
+  - Test/Make changes to a website without exposing commits to the public
+  - Safeguard against any potential leaks of sensitive information
+
+##### **Cons**
+  - Not able to utilize CodeQL for free
+  - Potentially out of compliance with any agency/program transparency initiative
+
+## **User Permissions**
+
+It is important to designate admin/owner access of the repository to an individual(s) who is either the persistent site owner or a federal employee with the “manager” role within the Pages organization. It is strongly recommended to not grant admin/owner privileges of the repository to any contractors or other temporary employees. This is such due to certain instances where actions will need to be taken by either Pages engineers or customers where it will be necessary to have an active user with admin access to a website's repository. Additionally when adding contributors/collaborators to a private repository it should mirror the member count that your organization has within Pages. Please note that removing an organization member from an organization within Pages will **not** automatically remove them from the repository within GitHub. They will still retain whatever read/write access they have in GitHub until explicitly removed.
+
+
+## **Security**
+
+### Dependabot
+
+Dependabot is a useful and highly configurable automated dependency management tool which assists developers with keeping third-party dependencies up-to-date with their latest version. It can be extremely beneficial to smaller agencies whose developer teams are limited or in some cases a single developer by reducing maintenance overhead. It scans the repository for outdated dependencies and generates pull requests against whatever branch it is enabled on to update them. Check out our [knowledge base article on Dependabot](https://cloud.gov/pages/knowledge-base/dependabot-with-pages/#configuring-dependabot) to view how to enable and configure it for your repository.
+
+### Branch Protection
+
+[Branch protection](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches) is a security feature within GitHub that limits users with write and push access the ability to push directly to any branch where branch protections are enabled. This includes but is not limited to requiring a pull request to merge code, setting the number of pull request reviews required, requiring status checks, dismissing approvals and other configuration options. With good branch protection settings you can protect your repository from an attack by a malicious actor with the stolen credentials from a single maintainer of that repository. 
```

Comment:
I think "that limits users with write and push access the ability to push directly" needs to be reworded for clarity.